pkg/k8sutil/pv.go

```diff
@@ -75,12 +75,22 @@ func UpdateReclaimPolicy(client *k8s.Clientset, pv *corev1.PersistentVolume) err
 	return err
 }
 
-func GetFlexVolumeName(pv *corev1.PersistentVolume) string {
+func GetVolumeName(pv *corev1.PersistentVolume) string {
 	// Rook creates rbd image with PV name
 	return pv.Name
 }
 
 func GetCSIVolumeName(pv *corev1.PersistentVolume) string {
+	retry := 0
+	maxRetry := 5
+	for pv.Spec.CSI.VolumeAttributes["imageName"] == "" && retry < maxRetry {
+		fmt.Printf("Waiting up to %v for PersistentVolume %q to be Created", retry, pv.Name)
+		if retry >= maxRetry-1 {
+			return ""
+		}
+		time.Sleep(time.Second * 2)
```

Comment:
Let's log something every time we sleep so they can see what the tool is doing.

Reply:
The function now tracks the required retries :+1: